enterprise/app/code/code.tsx

```diff
@@ -261,24 +269,28 @@ export default class CodeComponent extends React.Component<Props, State> {
     return this.parsePath().path;
   }
 
-  fetchCode() {
+  async fetchCode() {
     const storageValue = localStorage.getItem(this.getStateCacheKey());
     const storedState = storageValue ? (JSON.parse(storageValue, stateReviver) as State) : undefined;
     if (storedState) {
-      this.setState(storedState);
-      if (storedState.repoResponse && storedState.commitSHA) {
+      this.setState(storedState, () => {
+        repoLoaded(storedState.repoResponse);
+      });
+      if (storedState.treeResponse && storedState.commitSHA) {
         return;
       }
     }
     if (!this.currentOwner() || !this.currentRepo()) {
       return;
     }
 
-    let commit = this.state.commitSHA || "master";
+    let repoResponse = await this.octokit!.request(`GET /repos/${this.currentOwner()}/${this.currentRepo()}`);
+    repoLoaded(repoResponse);
+    let commit = this.state.commitSHA || repoResponse.data.default_branch;
     this.octokit!.request(`/repos/${this.currentOwner()}/${this.currentRepo()}/git/trees/${commit}`).then(
-      (response: any) => {
-        console.log(response);
-        this.updateState({ repoResponse: response, commitSHA: response.data.sha });
+      (treeResponse: any) => {
+        console.log(treeResponse);
+        this.updateState({ repoResponse: repoResponse, treeResponse: treeResponse, commitSHA: treeResponse.data.sha });
       }
```

Comment:
nit, the serial requests here will probably be slow (because the GH API is generally pretty slow) - if you want, you could do

```
let [repoResponse, treeResponse] = await Promise.all([
  this.octokit!.request(`GET /repos/${this.currentOwner()}/${this.currentRepo()}`),
  this.octokit!.request(`/repos/${this.currentOwner()}/${this.currentRepo()}/git/trees/${commit}`),
])
this.updateState({ reposResponse, treeResponse })

```

Reply:
You need the result from the first request to make the second request if the commitSha isn't specified, i.e.
```
    let commit = this.state.commitSHA || repoResponse.data.default_branch;
```